custom_components/hyxi_cloud/coordinator.py

```
@@ -147,6 +176,25 @@ async def _async_update_data(self):
+                if cached_devices:
+                    self.hyxi_metadata["last_error"] = str(err)
+                    self.hyxi_metadata["api_status"] = "Offline"
+                    _LOGGER.exception(
+                        "HYXI Cloud API fetch failed: %s. Falling back to %d cached devices from storage.",
+                        err,
+                        len(cached_devices),
+                    )
+                    self._merge_metrics(cached_devices)
+                    await self._async_sync_device_metadata(cached_devices)
+                    return cached_devices
+            except Exception as fallback_err:  # pylint: disable=broad-except
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)